integration/yarn/src/main/java/alluxio/yarn/ApplicationMaster.java

```diff
@@ -397,6 +410,37 @@ private void launchWorkerContainer(Container container) {
     }
   }
 
+  /**
+   * Checks if an Alluxio master node is already running
+   * or not on the master address given.
+   *
+   * @return true if master exists, false otherwise
+   */
+  private boolean masterExists() {
+
+    // TODO(dan): derive port from Constant
+    try {
+      URL myURL = new URL("http://" + mMasterAddress + ":" + "19999" + Constants.REST_API_PREFIX + "/master/version");
+      LOG.debug("Checking for master at: " + myURL.toString());
+      HttpURLConnection connection = (HttpURLConnection) myURL.openConnection();
+      connection.setRequestMethod(HttpMethod.GET);
+
+      int resCode = connection.getResponseCode();
+      LOG.debug("Response code from master was: " + Integer.toString(resCode));
+
+      if (resCode == HttpURLConnection.HTTP_OK) {
+        connection.disconnect();
+        return true;
+      }
+    } catch (MalformedURLException e) {
+      LOG.error("Malformed URL in attempt to check if master is running already", e);
+    } catch (IOException e) {
+      LOG.error("Unable to open connection to master in attempt to check if already running", e);
```

Comment:
Will this exception be thrown if there's no master running at the master address? If so this should be at the debug level since usually the master will not be running yet at this point.